errors.go

```diff
@@ -297,6 +302,33 @@ func FromGRPCStatus(grpcStatus status.Status) *AsertoError {
 	return result
 }
 
+/**
+ * Retrieves the most inner logger associated with an error.
+ */
+func Logger(err error) *zerolog.Logger {
+	var logger *zerolog.Logger
+
+	if err == nil {
+		return logger
+	}
+
+	for {
+		if ce, ok := err.(*ContextError); ok {
```

Comment:
I think this check should use `errors.As()` because `err` may not be a `ContextError` itself but it could be wrapping one.

```go
var ce *ContextError
if errors.As(err, &ctxErr) {
    if ctxLogger := extractLogger(ce.Ctx); ctxLogger != nil {
        logger = ctxLogger
    }
}